start.py

```diff
@@ -0,0 +1,337 @@
+            KDC_USER_KEYTAB_FILEPATH, ' '.join(principal_list))
+        kdc_commands.append(create_keytab_cmd)
+
+    kdc_commands.extend([
+        'krb5kdc',
+        'kadmind',
+        'authconfig --enablekrb5 --update'
+    ])
+
+    kdc_commands.append('cp -f {} {}'.format(KDC_KRB5_CONF_FILEPATH, KERBEROS_VOLUME_DIR))
+    if args.kerberos_principals:
+        kdc_commands.append('chmod 644 {}'.format(KDC_USER_KEYTAB_FILEPATH))
+
+    kdc_node.execute(command="bash -c '{}'".format('; '.join(kdc_commands)), quiet=quiet)
+
+    logger.info('Validating service health ...')
+    _validate_kdc_health(node=kdc_node, services=['krb5kdc', 'kadmin'], quiet=quiet)
+
+    # Add DSE specific logic on KDC host
+    logger.info('Creating `dse` and `HTTP` Kerberos principals for DSE nodes ...')
+    for node in nodes:
+        key_tab_filename = '{}/{}.keytab'.format(KERBEROS_VOLUME_DIR, node.fqdn)
+        kdc_dse_commands = [
+            'rm -f {}/{}.keytab'.format(KERBEROS_VOLUME_DIR, node.fqdn),
+            'kadmin.local -q "addprinc -randkey dse/{}"'.format(node.fqdn),
+            'kadmin.local -q "addprinc -randkey HTTP/{}"'.format(node.fqdn),
+            'kadmin.local -q "ktadd -k {} dse/{}"'.format(key_tab_filename, node.fqdn),
+            'kadmin.local -q "ktadd -k {} HTTP/{}"'.format(key_tab_filename, node.fqdn),
+            'chmod 644 {}'.format(key_tab_filename)
+        ]
+        kdc_node.execute(command="bash -c '{}'".format('; '.join(kdc_dse_commands)), quiet=quiet)
+
+    # DSE node logic
+    logger.info('Updating DSE configurations and starting DSE nodes ...')
+    cluster_name = args.dse_cluster_name
+    cluster_seeds = ','.join(node.ip_address for node in nodes)
+    for node in nodes:
+        # kerberos specific commands
+        dse_kdc_commands = [
+            'cp {}/krb5.conf /etc/.'.format(KERBEROS_VOLUME_DIR),
+            'cp {}/{}.keytab {}'.format(KERBEROS_VOLUME_DIR, node.fqdn, DSE_KEYTAB_FILEPATH),
+            'chown cassandra:cassandra {}'.format(DSE_KEYTAB_FILEPATH),
+            'chmod 600 {}'.format(DSE_KEYTAB_FILEPATH)
+        ]
+        if args.kerberos_principals:
+            dse_kdc_commands.extend([
+                'cp {} {}/.'.format(KDC_USER_KEYTAB_FILEPATH, DSE_HOME_DIR),
+                'chown cassandra:cassandra {}'.format(DSE_USER_KEYTAB_FILEPATH),
+                'chmod 600 {}'.format(DSE_USER_KEYTAB_FILEPATH)
+            ])
+        node.execute(command="bash -c '{}'".format('; '.join(dse_kdc_commands)), quiet=quiet)
+        # DSE config specific commands
+        dse_config_commands = [
+            'cp {} {}.org'.format(DSE_CASSANDRA_CONF_FILEPATH, DSE_CASSANDRA_CONF_FILEPATH),
+            'cp {} {}.org'.format(DSE_CONF_FILEPATH, DSE_CONF_FILEPATH),
+            'mkdir -p {}'.format(DSE_CQLSHRC_HOME_DIR)
+        ]
+        node.execute(command="bash -c '{}'".format('; '.join(dse_config_commands)), quiet=quiet)
+        # DSE cassandra.yaml mods
```

Comment:
Any way to abstract away some of the duplication between this and the previous mods defined elsewhere? Might wanna have a common function that does behavior common to both and then just defer to specific Kerberos or non-Kerberos behavior from there?